### ISSUE TITLE
[Frontend] ContentAttributes support ?

### ISSUE THREAD

**Issue by archyben** [DVBProject/DVB-I-Reference-Client#53]
Does the DVB-I reference client support DVB-I ContentAttributes , such as VideoConformancePoint element ?

Thanks in advance,

Benjamin

**Comment by paulhiggs** [comment]
No, the evaluation of specified ContentAttributes against device capabilities is not provided in the current release. Player capabilities need to be exposed to the application, but then it could become part of the service instance filtering algorithm.

**Comment by juhajoki** [comment]
We can add this to the backlog

**Comment by paulhiggs** [comment]
@bpeexc what would you expect from the DVB-I Reference Client if it were to example the `ContentAttributes`? 

Should it disable any DASH delivered service instances that cannot be played in the &lt;video&gt; element?

How can the DVB-I application 'know' what the supported broadcast receiver codecs are?

**Comment by jpiesing** [comment]
Assuming we want the reference app to stick to standard HTML5 (and omit HbbTV extensions) then it might make sense to discard broadband service instances where [MediaSource.isTypeSupported](https://www.w3.org/TR/media-source-2/#dom-mediasource-istypesupported ) returns false.

**Comment by paulhiggs** [comment]
Yes, we would need to define a mapping between DVB classification scheme terms and `type` for isTypeSupported().

is there some equivalent of isTypeSupported for the `video/broadcast` object or would any 'unplayable broadcast services' be removed during channel scan?

**Comment by jpiesing** [comment]
> is there some equivalent of isTypeSupported for the `video/broadcast` object or would any 'unplayable broadcast services' be removed during channel scan?

Normally the latter. Additionally, the broadcast codecs supported are listed in the XML capabilities as a list of <broadcast> elements whose value is one of those in either DVB VideoConformancePointsCS:2017 or a AudioConformancePointsCS:2017.

**Comment by paulhiggs** [comment]
I just stumbled upon the [`decodingInfo`](https://developer.mozilla.org/en-US/docs/Web/API/MediaCapabilities/decodingInfo) API which could be used in conjunction with `<VideoAttributes>` and `<AudioAttributes>` to determine if a service instance is playable.

**Comment by jpiesing** [comment]
> I just stumbled upon the [`decodingInfo`](https://developer.mozilla.org/en-US/docs/Web/API/MediaCapabilities/decodingInfo) API which could be used in conjunction with `<VideoAttributes>` and `<AudioAttributes>` to determine if a service instance is playable.

It does look widely supported - https://caniuse.com/mdn-api_mediacapabilities_decodinginfo - although how accurate the answers are remains to be seen.
